Security/Web_Security.mediawiki

```diff
@@ -0,0 +1,614 @@
+|-
+! Configuration
+! Oldest compatible clients
+|- style="background-color: #9EDB58;"
+| align="center" | Modern
+| style="padding-left: .5em;" | Firefox 27, Chrome 22, Internet Explorer 11, Opera 14, Safari 7, Android 4.4, Java 8 
+|- style="background-color: #E8E27A;"
+| align="center" | Intermediate
+| style="padding-left: .5em;" | Firefox 1, Chrome 1, Internet Explorer 7, Opera 5, Safari 1, Internet Explorer 8 (XP), Android 2.3, Java 7 
+|- style="background-color: #CCCCCC;"
+| align="center" | Backwards
+| style="padding-left: .5em;" | Internet Explorer 6 (XP), Java 6 
+|}
+
+=== See Also ===
+
+* [[Security/Server Side TLS|Mozilla Server Side TLS Guidelines]]
+* [https://mozilla.github.io/server-side-tls/ssl-config-generator/ Mozilla Server Side TLS Configuration Generator] - generates software configurations for the three levels of compatibility
+
+
+
+== HTTP Strict Transport Security ==
+
+HTTP Strict Transport Security (HSTS) is an HTTP header that notifies user agents to only connect to a given site over HTTPS, even if the scheme chosen was HTTP.  Browsers that have had HSTS set for a given site will transparently upgrade all requests to HTTPS.  HSTS also tells the browser to treat TLS and certificate-related errors more strictly by disabling the ability for users to bypass the error page.
+
+The header consists of one mandatory parameter (<tt>max-age</tt>) and two optional parameters (<tt>includeSubDomains</tt> and <tt>preload</tt>), separated by semicolons.
+
+=== Directives ===
+
+* <tt>max-age:</tt> how long user agents will redirect to HTTPS, in seconds
+* <tt>includeSubDomains:</tt> whether user agents should upgrade requests on subdomains
+* <tt>preload:</tt> whether the site should be included in the [https://hstspreload.appspot.com/ HSTS preload list]
+
+<tt>max-age</tt> must be set to a minimum of six months (15768000), but longer periods such as one year (31536000) or longer are also acceptable.  Note that once this value is set, the site must continue to support HTTPS until the expiry time has been reached.
+
+<tt>includeSubDomains</tt> notifies the browser that all subdomains of the current origin should also be upgraded via HSTS.  For example, setting <tt>includeSubDomains</tt> on <tt>domain.mozilla.com</tt> will also set it on <tt>host1.domain.mozilla.com</tt> and <tt>host2.domain.mozilla.com</tt>. Extreme care is needed when setting the <tt>includeSubDomains</tt> flag, as it could disable sites on subdomains that don't yet have HTTPS enabled.
+
+<tt>preload</tt> allows the website to be included in the [https://hstspreload.appspot.com/ HSTS preload list], upon submission. As a result, web browsers will do HTTPS upgrades to the site without ever having to receive the initial HSTS header.  This prevents downgrade attacks upon first use and is recommended for all high risk websites.  Note that being included in the HSTS preload list requires that <tt>includeSubDomains</tt> also be set.
+
+=== Examples ===
```

Comment:
l68: which one should I set? the standard has to show the user exactly what to do, specially since they will only read the description if they have no clue whats going on. Most of the time they need something to copy paste that will provide their expected level of security